tests/install_upgrade_operators/network_policy/conftest.py

```diff
@@ -0,0 +1,186 @@
+import shlex
+
+import pytest
+from ocp_resources.network_policy import NetworkPolicy
+from ocp_resources.pod import Pod
+from ocp_resources.service import Service
+
+from utilities.constants import POD_CONTAINER_SPEC, POD_SECURITY_CONTEXT_SPEC
+
+from .utils import TEST_SERVER_APP_LABEL, TEST_SERVER_PORT, AllowAllNetworkPolicy
+
+
+@pytest.fixture
+def network_policy_by_name(admin_client, hco_namespace, network_policy_name):
+    return NetworkPolicy(
+        client=admin_client, namespace=hco_namespace.name, name=network_policy_name, ensure_exists=True
+    )
+
+
+@pytest.fixture
+def deployed_client_pod(admin_client, hco_namespace):
+    with Pod(
+        name="network-policy-client-pod",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        security_context=POD_SECURITY_CONTEXT_SPEC,
+        containers=[POD_CONTAINER_SPEC],
+    ) as pod:
+        pod.wait_for_status(status=Pod.Status.RUNNING)
+        yield pod
+
+
+@pytest.fixture
+def deployed_server_pod(admin_client, hco_namespace):
+    server_container_spec = {
+        **POD_CONTAINER_SPEC,
+        "name": "test-server",
+        "command": [
+            "/bin/bash",
+            "-c",
+            f'while true; do echo -e "HTTP/1.1 200 OK\\n\\nServer response $(date)" | '
+            f"nc -l -p {TEST_SERVER_PORT}; done",
+        ],
+        "ports": [{"containerPort": TEST_SERVER_PORT, "protocol": "TCP"}],
+    }
+
+    with Pod(
+        name="network-policy-server-pod",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        security_context=POD_SECURITY_CONTEXT_SPEC,
+        containers=[server_container_spec],
+        label={"app": TEST_SERVER_APP_LABEL},
+    ) as pod:
+        pod.wait_for_status(status=Pod.Status.RUNNING)
+        yield pod
+
+
+@pytest.fixture
+def deployed_server_service(admin_client, hco_namespace, deployed_server_pod):
+    with Service(
+        name="network-policy-server-service",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        selector={"app": TEST_SERVER_APP_LABEL},
+        ports=[{"name": "http", "port": TEST_SERVER_PORT, "targetPort": TEST_SERVER_PORT, "protocol": "TCP"}],
+    ) as service:
+        yield service
+
+
+@pytest.fixture
+def network_policy_match_labels(network_policy_by_name):
+    match_labels = network_policy_by_name.instance.spec.get("podSelector", {}).get("matchLabels")
+    assert match_labels, f"NetworkPolicy {network_policy_by_name.name} has no/empty podSelector.matchLabels"
+    return dict(match_labels)
```

Comment:
```suggestion
    return match_labels
```
IIRC, matchLabels is always dict

Reply:
I added but didn't help.

Reply:
not sure why you get the NoneType error.. you have assert which checks that the match_labels is set
also, it contains items when called directly:

```
(Pdb) p match_labels.items()
dict_items([('app', 'kubevirt-hyperconverged'), ('app.kubernetes.io/component', 'kubevirt-console-plugin')])
```

Reply:
yeah, me neither but I think the conversion in the fixture is elegant enough, right? It will return a dict so it should do the work for the ones making use of it.